internal/addon/authentication/provider.go

```diff
@@ -190,3 +232,24 @@ func BuildAuthenticationMap(inputMap map[string]string) map[Target]Authenticatio
 
 	return result
 }
+
+func BuildAuthenticationFromAnnotations(annotations map[string]string) (map[Target]AuthenticationType, error) {
+	result := make(map[Target]AuthenticationType)
+	for annotation, annValue := range annotations {
+		if !strings.HasPrefix(annotation, AnnotationAuthOutput) {
+			continue
+		}
+		split := strings.Split(annotation, "/")
+		if len(split) != 2 {
+			return result, kverrors.New("unable to extract output name from annotation", "annotation", annotation)
+		}
+		if split[1] == "" {
+			return result, kverrors.New("output name not specified", "annotation", annotation)
+		}
+		target := Target(split[1])
+		authType := AuthenticationType(annValue)
+		result[target] = authType
+	}
+
+	return result, nil
+}
```

Comment:
Why is this function as well as the `BuildAuthenticationMap` part of the public interface of the package when used only by the `GenerateSecrets` function? Suggest to make both private and use them in `GenerateSecrets` only for now by adapting it's interface as such:

```go
func (sp *secretsProvider) GenerateSecrets(ctx context.Context, annotations map[string]string) (map[Target]SecretKey, error) {
```